app/controllers/AtomController.scala

```diff
@@ -35,12 +35,11 @@ trait AtomController extends Controller with UnpackedDataStores {
 
   import authActions.APIAuthAction
 
-  object CanUploadAsset extends ActionBuilder[UploadUserRequest] {
+  object ActionWithPermission extends ActionBuilder[UploadUserRequest] {
```

Comment:
Yeah we struggled with the naming too. It won't block anyone based on a particular permission but will pass all the permissions to the next block so that can make a decision.

We also tried pulling out the code to be a simple function rather than an `ActionBuilder` but then we would have had to refactor the parsing helper method to take `Future[Result]` as well. Messy stuff.

Janus calls them `APIAuthAction`, perhaps `APIPermissionAction` is reasonable to follow the same convention?